client/shipping-label/state/reducer.js

```diff
@@ -328,25 +329,39 @@ reducers[ OPEN_ADD_ITEM ] = ( state ) => {
 	return {
 		...state,
 		showAddItemDialog: true,
+		addedItems: {},
 	};
 };
 
 reducers[ CLOSE_ADD_ITEM ] = ( state ) => {
 	return {
 		...state,
-		movedItemIndex: -1,
 		showAddItemDialog: false,
 	};
 };
 
-reducers[ SET_ADDED_ITEM ] = ( state, { sourcePackageId, movedItemIndex } ) => {
+reducers[ SET_ADDED_ITEM ] = ( state, { sourcePackageId, movedItemIndex, added } ) => {
 	return {
 		...state,
-		sourcePackageId,
-		movedItemIndex,
+		addedItems: { ...state.addedItems,
+			[ sourcePackageId ]: added
+				? { ...state.addedItems[ sourcePackageId ], [ movedItemIndex ]: true }
+				: _.omit( state.addedItems[ sourcePackageId ], movedItemIndex )
```

Comment:
How much cleaner would this be if we weren’t using the item index as an object property?
 
`{ addedItems: { 'package_1': [ 0, 3, 4 ] } }` array instead?